app/models/locations/map.rb

```diff
@@ -0,0 +1,99 @@
+module Locations
+  class Map
+    attr_accessor :id,:loc
+
+    def initialize(id: nil, loc: nil)
+      @locator_url = "http://library.princeton.edu/locator/index.php?"
+      @stackmap_url = "http://princeton.stackmap.com/view/?"
+      @id ||= id
+      @loc ||= loc
+    end
+
+    def url
+      if is_valid?
+        if self.locator_libs.include? self.lib.code
+          @locator_url + "loc=" + self.loc + "&id=" + self.id
+        elsif self.stackmap_libs.include? self.lib.code
+          if !self.closed_stack_reserves.include? self.loc
+            if self.by_title_locations.include? self.loc
+              callno = self.bibrec['title_sort'].first
+            else
+              callno = self.bibrec['call_number_display'].first
+            end
+            URI.encode(@stackmap_url + "callno=" + callno  + "&location=" + self.loc + "&library=" + self.lib.label)
+          else
```

Comment:
@sdellis I don't see a test for the reserves use case. Sorry if I missed it. I think added this would make coveralls happy.